src/compiler/parser.ts

```diff
@@ -4415,7 +4447,7 @@ namespace ts {
             while (parseOptional(SyntaxKind.DotToken)) {
                 const propertyAccess: JsxTagNamePropertyAccess = <JsxTagNamePropertyAccess>createNode(SyntaxKind.PropertyAccessExpression, expression.pos);
                 propertyAccess.expression = expression;
-                propertyAccess.name = parseRightSideOfDot(/*allowIdentifierNames*/ true);
+                propertyAccess.name = parseRightSideOfDot(/*allowIdentifierNames*/ true, /*allowPrivateIdentifiers*/ true);

```

Comment:
We allow private identifiers here, but I don't see any tests for tsx/jsx input.

Reply:
Added: privateNameJsx.tsx

Reply:
I've updated this so that private identifiers are not allowed in JSX tag names because they are not in [the JSX spec](https://facebook.github.io/jsx/).